PerfettoCds/Pipeline/CompositeDataCookers/PerfettoProcessMemoryEventCooker.cs

```diff
@@ -0,0 +1,132 @@
+
+        public DataCookerPath Path => DataCookerPath;
+
+        // Declare all of the cookers that are used by this CompositeCooker.
+        public IReadOnlyCollection<DataCookerPath> RequiredDataCookers => new[]
+        {
+            PerfettoPluginConstants.CounterCookerPath,
+            PerfettoPluginConstants.ProcessCounterTrackCookerPath,
+            PerfettoPluginConstants.ProcessCookerPath
+        };
+
+        [DataOutput]
+        public ProcessedEventData<PerfettoProcessMemoryEvent> ProcessMemoryEvents { get; }
+
+        public PerfettoProcessMemoryEventCooker() : base(PerfettoPluginConstants.ProcessMemoryEventCookerPath)
+        { 
+            this.ProcessMemoryEvents =
+                new ProcessedEventData<PerfettoProcessMemoryEvent>();
+        }
+
+        public void OnDataAvailable(IDataExtensionRetrieval requiredData)
+        {
+            // Gather the data from all the SQL tables
+            var counterData = requiredData.QueryOutput<ProcessedEventData<PerfettoCounterEvent>>(new DataOutputPath(PerfettoPluginConstants.CounterCookerPath, nameof(PerfettoCounterCooker.CounterEvents)));
+            var processCounterTrackData = requiredData.QueryOutput<ProcessedEventData<PerfettoProcessCounterTrackEvent>>(new DataOutputPath(PerfettoPluginConstants.ProcessCounterTrackCookerPath, nameof(PerfettoProcessCounterTrackCooker.ProcessCounterTrackEvents)));
+            var processData = requiredData.QueryOutput<ProcessedEventData<PerfettoProcessEvent>>(new DataOutputPath(PerfettoPluginConstants.ProcessCookerPath, nameof(PerfettoProcessCooker.ProcessEvents)));
+
+            // Join them all together
+            // Counter table contains the memory count value, timestamp
+            // ProcessCounterTrack contains the UPID and memory type name. All the memory types we care about start with "mem."
+            // Process contains the process name
+            var joined = from counter in counterData
+                         join processCounterTrack in processCounterTrackData on counter.TrackId equals processCounterTrack.Id
+                         join process in processData on processCounterTrack.Upid equals process.Upid
+                         where processCounterTrack.Name.StartsWith("mem.")
+                         orderby counter.Timestamp ascending
+                         select new { counter, processCounterTrack, process };
+
+            // Create events out of the joined results
+            foreach (var processGroup in joined.GroupBy(x => x.processCounterTrack.Upid))
+            {
+                var timeGroups = processGroup.GroupBy(z => z.counter.RelativeTimestamp);
+
+                for (int i = 0; i < timeGroups.Count(); i++)
+                {
+                    var timeGroup = timeGroups.ElementAt(i);
+
+                    var ts = timeGroup.Key;
+                    var processName = timeGroup.ElementAt(0).process.Name + " " + processGroup.Key;
```

Comment:
NIT: $"{timeGroup.ElementAt(0).process.Name} {processGroup.Key}" is a bit cleaner